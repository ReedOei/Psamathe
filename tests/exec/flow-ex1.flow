contract Main {}

;

var A : ! nat := 0
var B : ! nat := 0
var C : ! nat := 0
var D : any set ! string
var E : any set ! string

2 --> A
if A != 2 { revert("Failure!") }

A --[ 1 ]-> B
B --> C

"foo" --> D
"bar" --> D
if "bar" not in D { revert("Failure!") }

D --[ "foo" ]-> E
if "foo" in D { revert("Failure!") }

A --> consume
if A > 0 { revert("Failure!") }

