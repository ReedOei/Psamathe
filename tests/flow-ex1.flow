contract Main {}

;

var A : ! nat
var B : ! nat
var C : ! nat
var D : any set ! string
var E : any set ! string

2 --> A
A --[ 1 ]-> B
B --> C

A --[ 1 ]-> C --> B

"foo" --> D
"bar" --> D
D --[ "foo" ]-> E

A --> consume
B --[ 1 ]-> consume

