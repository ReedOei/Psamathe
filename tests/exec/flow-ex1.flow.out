<generatedTop>
  <common>
    <k>
      .
    </k>
    <contracts>
      Main |-> contract Main { .Decls }
    </contracts>
    <fields>
      Main |-> .Set
    </fields>
    <constructors>
      Main |-> constructor ( .VarDefs , pass )
    </constructors>
    <transactions>
      .Map
    </transactions>
    <types>
      address |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , address )
      bool |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , bool )
      nat |-> modified ( SetItem ( consumable )
      SetItem ( fungible ) , nat )
      string |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , string )
    </types>
    <freshCounter>
      8
    </freshCounter>
  </common>
  <lookup>
    this |-> loc ( -1 )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( nat , 0 )
    loc ( 1 ) |-> resource ( nat , 0 )
    loc ( 2 ) |-> resource ( nat , 1 )
    loc ( 3 ) |-> resource ( set ! string , SetItem ( "bar" ) )
    loc ( 4 ) |-> resource ( set ! string , SetItem ( "foo" ) )
    loc ( 5 ) |-> resource ( nat , 0 )
    loc ( 6 ) |-> resource ( string , emptyval )
    loc ( 7 ) |-> resource ( string , emptyval )
    loc ( -1 ) |-> resource ( Base , .Map )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
